argilla/docs/how_to_guides/webhooks.md

```diff
@@ -0,0 +1,162 @@
+---
+
+# Webhooks
+
+In this section, we will provide a step-by-step guide to create a webhook in Argilla.
+
+## What is a webhook?
+
+A webhook is a way for an app to provide other applications with real-time information. A webhook delivers data to other applications as it happens, meaning you get data immediately. Unlike typical APIs where you would need to poll for data very frequently in order to get it real-time.
+This makes webhooks much more efficient for both provider and consumer.
+
+## How to create a webhook in Argilla?
+
+The python SDK provides a simple way to create a webhook in Argilla. Its allow you focus on the use case of the webhook
+and not on the implementation details. You only need to create your event handler function with the `webhook_listener` decorator.
+
+```
+import argilla as rg
+
+from datetime import datetime
+from argilla_sdk import webhook_listener
+
+@webhook_listener(events="dataset.created")
+async def my_webhook_handler(dataset: rg.Dataset, type: str, timestamp: datetime):
+    print(event)
+```
+
+In the example above, we have created a webhook that listens to the `dataset.created` event. (You can find the list of events in the [Events](#events) section).
+
+The python SDK will automatically create a webhook in Argilla and listen to the specified event. When the event is triggered,
+the `my_webhook_handler` function will be called with the event data. The SDK will also parse the incoming webhook event into
+a proper resource object (rg.Dataset, rg.Record, and rg.UserResponse). The SDK will also take care of request authentication and error handling.
+
+## How to run the webhook?
+
+Under the hood, the SDK uses the `FastAPI` framework to create the webhook server and the POST endpoint to receive the webhook events.
+
+To run the webhook, you need to define the webhook server in your code and start it using the `uvicorn` command.
+
+```
+from argilla import get_webhook_server
+
+server = get_webhook_server()
+```
+
+```bash
+uvicorn my_webhook:server
+```
+
+You can explore the Swagger UI to explore your defined webhooks by visiting `http://localhost:8000/docs`.
+
+
+The `uvicorn` command will start the webhook server on the default port `8000`.
+
+By default, the Python SDK will register the webhook using the server URL `http://127.0.0.1:8000/`. If you want to use a different server URL, you can set the `WEBHOOK_SERVER_URL` environment variable.
+
+```bash
+export WEBHOOK_SERVER_URL=http://my-webhook-server.com
+```
+
+All incoming webhook events will be sent to the specified server URL.
+
+## Webhooks management
+
+The Python SDK provides a simple way to manage webhooks in Argilla. You can create, list, update, and delete webhooks using the SDK.
+
+### Create a webhook
+
+You can create a webhook using the `create_webhook` method.
+
+```python
+import argilla as rg
+
+client = rg.Argilla(...)
+
+webhook = rg.Webhook(
+    url="https://my-webhook-server.com",
```

Comment:
```suggestion
    url="https://my-webhook-server.com",
```
maybe the ULRs are a bit confusing? should we perhaps use the WEBHOOK_SERVER_URL directly?